pkg/apis/wksprovider/controller/wksctl/machine_actuator.go

```diff
@@ -525,20 +528,28 @@ func (a *MachineActuator) update(ctx context.Context, cluster *clusterv1.Cluster
 			return err
 		}
 		contextLog.Infof("Master needs update: %t", masterNeedsUpdate)
+		updatingNode, err := a.findUpdatingNode()
+		if err != nil {
+			return err
+		}
+		if updatingNode != nil && isMaster(updatingNode) && updatingNode != node {
+			contextLog.Infof("Master is updating: %s", updatingNode.Name)
+			return fmt.Errorf("Master %s is currently updating...", updatingNode.Name)
+		}
 		isOriginal, err := a.isOriginalMaster(node)
 		if err != nil {
 			return err
 		}
 		contextLog.Infof("Is original: %t", isOriginal)
-		if (!isOriginal && originalNeedsUpdate) || (!isMaster && masterNeedsUpdate) {
+		if (!isOriginal && originalNeedsUpdate) || (!nodeIsMaster && masterNeedsUpdate) || (updatingNode != nil && updatingNode != node) {
 			return errors.New("Master nodes must be upgraded before worker nodes")
```

Comment:
This message will be misleading if the updatingNode is a worker and it isn't this node.